cinema/views.py

```diff
@@ -34,6 +36,23 @@
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        actor_names = self.request.query_params.get("actor")
+        genre_names = self.request.query_params.get("genre")
+        title = self.request.query_params.get("title")
+        if actor_names:
+            actor_list = actor_names.split(",")
+            queryset = queryset.filter(actors__last_name__in=actor_list)
```

Comment:
The filtering logic for actors is incorrect. It should filter by full names, not by last names. Consider using `actors__full_name__in=actor_list`.